### PR TITLE
Strengthen TypeScript CI and release gates

### LINKED ISSUE #22: Strengthen TypeScript CI and release gates
## Context

Recent `crap-typescript` CI/release issues apply to `cognitive-typescript` because the sibling projects should have comparable validation strength for supported runtime versions and platform-sensitive paths.

Related `crap-typescript` issues:

- fabian-barney/crap-typescript#125
- fabian-barney/crap-typescript#126
- fabian-barney/crap-typescript#127
- fabian-barney/crap-typescript#131

## Task

Strengthen CI and release workflows without adding coverage collection or CRAP-specific behavior.

CI requirements:

- Test the oldest supported Node version from `package.json` engines. If engines remain `>=18`, CI must include Node `18`.
- Also test active LTS/current Node versions used by the sibling projects; at minimum include Node `22` and `24`, and add the current release line if the project supports it.
- Add Windows coverage for platform-sensitive behavior, especially report path validation, file selection, changed-file detection, and filesystem case handling.
- Keep Linux coverage for the standard build/test/publish validation path.
- Reduce duplicate `npm ci` work where practical without weakening job isolation; use a clear matrix or dependency cache strategy.

Release requirements:

- `release.yml` must run the `cognitive-typescript` gate and the sibling `crap-typescript` gate before publishing.
- Release validation must fail before publication if either gate fails.
- Release workflow must use the same documented Node/npm compatibility assumptions as CI.

## Non-goals

- Do not force coverage, run coverage reporters, parse Istanbul/LCOV, or add CRAP score computation.
- Do not remove useful job isolation just to avoid repeated installs.

## Acceptance Criteria

- CI matrix covers Node `18` if still supported, plus active LTS/current Node versions used by the repo.
- CI has Windows coverage for path-sensitive code.
- Release workflow runs cognitive and CRAP gates before npm publish.
- Workflow dependency installation is not unnecessarily duplicated where a cache or matrix can solve it cleanly.
- README/CONTRIBUTING list the same local validation commands the workflows enforce.
- Tests/workflow checks prove report path validation and changed-file detection run on Windows.

### PR DESCRIPTION
**Pull request by fabian-barney**
Closes #22

Supersedes #32 so GitHub can run a fresh Copilot review on the already-validated final head.

## Summary
- add Ubuntu and Windows CI coverage for Node 22.12.0 and 24.0.0
- keep cognitive and CRAP gates explicit in CI, with isolated CRAP gate installs instead of fragile cross-job workspace handoff
- split release validation from privileged publishing and require both gates before publication

## Validation
- npm ci
- npm run build
- npm test
- npm run cognitive-typescript-check
- npm run crap-typescript-check
- npm pack --workspaces
- npm run verify-release-version -- v0.1.1

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -39,17 +89,41 @@ jobs:
...
         uses: actions/setup-node@v6
         with:
-          node-version: "24"
+          node-version: ${{ matrix.node-version }}
+
+      - name: Set up npm 11.14.1
+        run: npm install --global npm@11.14.1
+
+      - name: Run cognitive-typescript gate
+        run: node ./scripts/run-cognitive-typescript-gate.mjs packages
```
[the remark this answers is not in the document]
Comment (fabian-barney) [reply .github/workflows/build.yml]: Fixed. The gate helper now runs the locally built CLI from `packages/cli/dist/bin.js`, and the CI gate job installs dependencies and builds the workspace before invoking `npm run cognitive-typescript-check`. That removes the hard-coded npm package version dependency and validates the branch artifacts directly.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -39,29 +37,71 @@ jobs:
...
+        run: npm run build
+
+      - name: Run tests
+        run: node ./node_modules/vitest/vitest.mjs run
+
+      - name: Run cognitive-typescript gate
+        run: npm run cognitive-typescript-check
+
+      - name: Run crap-typescript gate
+        run: node ./node_modules/vitest/vitest.mjs run --config vitest.crap.config.ts
```
[the remark this answers is not in the document]
Comment (fabian-barney) [reply .github/workflows/release.yml]: Fixed. The release validation job now runs the documented npm entrypoints: `npm test`, `npm run cognitive-typescript-check`, and `npm run crap-typescript-check`. That keeps the workflow aligned with the README and avoids bypassing future script logic.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -39,29 +37,71 @@ jobs:
...
+
+      - name: Build publish artifacts
+        run: |
+          mkdir -p release-artifacts
+          npm pack --workspaces --pack-destination release-artifacts
+          cp release-notes.md release-artifacts/release-notes.md
+
```
[the remark this answers is not in the document]
Comment (fabian-barney) [reply .github/workflows/release.yml:54]: Fixed. `npm run cognitive-typescript-check` no longer shells out to a hard-coded published CLI version. The helper script now executes the locally built CLI after the workflow builds the workspace, so tag-triggered releases validate the exact artifacts under test without waiting for npm publication.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -11,3 +11,22 @@ All changes in this repository are expected to be issue-linked.
...
+## Local validation
+
+Run the full Ubuntu validation command set before pushing. CI also runs Windows build/test coverage separately to catch path and platform issues:
+
+```bash
+npm ci
+npm run build
+npm test
+npm run cognitive-typescript-check
+npm run crap-typescript-check
```
[the remark this answers is not in the document]
Comment (fabian-barney) [reply CONTRIBUTING.md:23]: Already fixed on the current head. `npm run cognitive-typescript-check` now executes the locally built workspace CLI from `packages/cli/dist/bin.js`, so the local validation command exercises the branch artifacts instead of a pinned published npm version.

File: README.md [hunk 5]
```diff
@@ -108,7 +110,7 @@ Verified and unverified syntax shapes are tracked in [docs/compatibility-matrix.
 
 ## Release
 
-The default release path uses npm Trusted Publishing from `.github/workflows/release.yml`. Tag `v<version>` from `main` after the build workflow is green. The tag-triggered release workflow verifies package versions, renders the GitHub release notes from `CHANGELOG.md`, publishes the four public npm packages, and creates the GitHub release.
+The default release path uses npm Trusted Publishing from `.github/workflows/release.yml`. Tag `v<version>` from `main` after the build workflow is green. The tag-triggered release workflow verifies package versions, renders the GitHub release notes from `CHANGELOG.md`, runs `npm test`, `npm run cognitive-typescript-check`, and `npm run crap-typescript-check`, then publishes the four public npm packages and creates the GitHub release.
```
[the remark this answers is not in the document]
Comment (fabian-barney) [reply README.md:113]: Already fixed on the current head. `.github/workflows/release.yml` now invokes `npm test` and `npm run crap-typescript-check`, so this README sentence matches the workflow again.

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -39,29 +37,68 @@ jobs:
...
-      - name: Run build and tests
+      - name: Run tests
         run: npm test
 
-      # Trusted Publishing uses the job OIDC token for auth and npm
-      # automatically emits provenance for public packages from public repos.
-      - name: Publish core
-        run: npm publish --workspace packages/core --access public
+      - name: Run cognitive-typescript gate
+        run: npm run cognitive-typescript-check
```
[the remark this answers is not in the document]
Comment (fabian-barney) [reply .github/workflows/release.yml:45]: Already fixed on the current head. The release validation job now runs `npm test`, and the separate explicit build step was removed.

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -39,29 +37,68 @@ jobs:
...
 
-      - name: Publish CLI
-        run: npm publish --workspace packages/cli --access public
+      - name: Run crap-typescript gate
+        run: npm run crap-typescript-check
 
-      - name: Publish Vitest adapter
-        run: npm publish --workspace packages/vitest --access public
+      - name: Build publish artifacts
+        run: |
```
[the remark this answers is not in the document]
Comment (fabian-barney) [reply .github/workflows/release.yml:50]: Already fixed on the current head. The release gate still uses `npm run cognitive-typescript-check`, but that helper now runs the locally built workspace CLI instead of a hard-coded published package version.

File: .github/workflows/build.yml [hunk 8]
```diff
@@ -58,21 +112,49 @@ jobs:
       - name: Install dependencies
```
[the remark this answers is not in the document]
Comment (fabian-barney) [reply .github/workflows/build.yml:112]: Already fixed on the current head. The build workflow now installs dependencies, builds the workspace, and runs `npm run cognitive-typescript-check`, which in turn uses the locally built CLI.

**Comment by fabian-barney** [comment]
Superseded by #36. GitHub stopped advancing this PR against the live branch head, so new pushes, checks, and review requests were no longer attached to the actual branch tip.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -11,3 +11,22 @@ All changes in this repository are expected to be issue-linked.
 5. Keep the pull request green, reply to review comments, and resolve threads only after the fix or an explicit invalidation response.
 6. Merge only after the latest review is newer than the latest push and all required checks are green.
 
+## Local validation
+
+Run the full Ubuntu validation command set before pushing. CI also runs Windows build/test coverage separately to catch path and platform issues:
+
+```bash
+npm ci
+npm run build
```
[the remark this answers is not in the document]
Comment (fabian-barney) [reply CONTRIBUTING.md:20]: Classification: invalid for the current code. The root `package.json` defines `npm test` as `vitest run --coverage.enabled true`; it does not invoke `npm run build`. The explicit build step in `CONTRIBUTING.md` is therefore not duplicate work under the current scripts, so no documentation change is needed.
